### PR TITLE
overload acceptBlock

### PR DESCRIPTION
**Pull request by Mr-Leshiy**

### CODE REVIEW

**Comment by Warchant** [comment]
Please fix CI

File: include/veriblock/entities/context.hpp [hunk 1]
```diff
@@ -11,19 +11,57 @@ namespace altintegration {
...
+  VbkBlock getContainingBlock() const;
+
+  /**
+   * Return true if contains endorsement data
+   * @return true if contains endorsement data
+   */
+  bool containsEndorsements() const;
+
+  /**
+   * Return BtcEndorsement of the PartialVbk
```
Comment (blackyblack) [inline include/veriblock/entities/context.hpp:40]: of the PartialVTB

File: include/veriblock/entities/context.hpp [hunk 2]
```diff
@@ -11,19 +11,57 @@ namespace altintegration {
...
+
+  /**
+   * Return BtcEndorsement id of the endorsement
+   * @return endorsement.id BtcEndorsement::id_t
+   */
+  typename BtcEndorsement::id_t getEndorsementId() const;
+
+  // generate without VBK context blocks, only BtcEndorsement VbkBlock
+  // containing and btc blocks
+  static PartialVTB generateFromVtb(const VTB& vtb);
```
Comment (blackyblack) [inline include/veriblock/entities/context.hpp:53]: If PartialVTB does not contain VBK context blocks why do you have context_vbk in the struct?

File: src/entities/context.cpp [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  std::transform(vtbs.begin(),
+                 vtbs.end(),
+                 std::back_inserter(p_vtbs),
+                 [&](const VTB& vtb) -> PartialVTB {
+                   PartialVTB p_vtb = PartialVTB::generateFromVtb(vtb);
+                   for (const auto& b : vtb.context) {
+                     p_vtb.context_vbk.push_back(std::make_shared<VbkBlock>(b));
+                   }
+                   return p_vtb;
+                 });
```
Comment (blackyblack) [inline src/entities/context.cpp:35]: It does not look safe. When generating PartialVTBs from the vector, you populate PartialVTB.context_vbk, but when you generate PartialVTB from the single VTB, context_vbk stays uninitialised.

File: include/veriblock/entities/context.hpp (hunk 2 above)
Comment (Mr-Leshiy) [reply include/veriblock/entities/context.hpp:53]: This context blocks fills independently in the other place, and that context can be different with the context blocks from the VTB, so in this place I don not copy them.

File: src/entities/context.cpp [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  return endorsement.id;
+}
+
+bool operator==(const PartialVTB& a, const PartialVTB& b) {
+  bool is_equal = a.btc.size() == b.btc.size();
+  for (size_t i = 0; i < a.btc.size() && is_equal; ++i) {
+    if (*a.btc[i] != *b.btc[i]) {
+      is_equal = false;
+    }
+  }
```
Comment (Warchant) [inline src/entities/context.cpp:56]: I think it should be enough to use regular ==

File: src/entities/context.cpp [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    p_vtb.btc.push_back(std::make_shared<BtcBlock>(b));
+  }
+  p_vtb.btc.push_back(std::make_shared<BtcBlock>(vtb.transaction.blockOfProof));
+
+  p_vtb.containing = std::make_shared<VbkBlock>(vtb.containingBlock);
+
+  return p_vtb;
+}
+
+std::vector<PartialVTB> PartialVTB::generateFromVtb(
```
Comment (Warchant) [inline src/entities/context.cpp:21]:
```suggestion
std::vector<PartialVTB> PartialVTB::fromVTB(
```

File: src/entities/vtb.cpp [hunk 6]
```diff
@@ -43,8 +43,12 @@ std::vector<uint8_t> VTB::toVbkEncoding() const {
 }
 
 VTB::id_t VTB::getId() const {
-  auto rawBytes = toVbkEncoding();
-  return sha256(rawBytes);
+  WriteStream stream;
+  transaction.toVbkEncoding(stream);
+  merklePath.toVbkEncoding(stream);
```
Comment (Warchant) [inline src/entities/vtb.cpp:48]: VTB should not have merkle path as part of ID

**Pull request merged**